go/spec/absolute_path.go

```diff
@@ -18,9 +18,9 @@ import (
 var datasetCapturePrefixRe = regexp.MustCompile("^(" + datas.DatasetRe.String() + ")")
 
 type AbsolutePath struct {
-	dataset string
-	hash    hash.Hash
-	path    types.Path
+	Dataset string
```

Comment:
I'm not sure what you mean. It seems like commenting the fields would be a little overkill on this and we have plenty of places where public types and/or their fields are not documented and they show up in godocs. Do we have a requirement that all public fields in public structs need doc? If so, I'll do it.